File: events.dart
import 'dart:convert';
import 'dart:io';

class Event {
  late final int id;
  late final int month;
  late final String monthNameAr;
  late final String monthNameEn;
  late final int day;
  late final int? yearOfOccurrence;
  late final String name;
  late final String nameEn;
  late final String description;
  late final String importance;
  late final String typeAr;
  late final String typeEn;

  Event(Map json) {
    id = json[idKey];
    month = json[monthKey];
    monthNameAr = json[monthNameArKey];
    monthNameEn = json[monthNameEnKey];
    day = json[dayKey];
    yearOfOccurrence = json[yearOfOccurrenceKey];
    name = json[nameKey];
    nameEn = json[nameEnKey];
    description = json[descriptionKey];
    importance = json[importanceKey];
    typeAr = json[typeArKey];
    typeEn = json[typeEnKey];
  }

  static const String idKey = 'id';
  static const String monthKey = 'month';
  static const String monthNameArKey = 'month_name_ar';
  static const String monthNameEnKey = 'month_name_en';
  static const String dayKey = 'day';
  static const String yearOfOccurrenceKey = 'year_of_occurrence';
  static const String nameKey = 'name';
  static const String nameEnKey = 'name_en';
  static const String descriptionKey = 'description';
  static const String importanceKey = 'importance';
  static const String typeArKey = 'type_ar';
  static const String typeEnKey = 'type_en';
}

class Events {
  late final List<Event> events;

  Events() {
    //    events = _readEvents();
  }

  Future<void> initEvents() async {
    final File file = File('events.json');
    final content = await file.readAsString();
    final json = jsonDecode(content);
    final List list = json['islamic_events'];
    events = list.map((e) => Event(e)).toList();
  }
}

void main(List<String> args) async {
  // final File file = File('events.json');
  // final content = await file.readAsString();
  // final json = jsonDecode(content);
  // final List list = json['islamic_events'];
  // final events = list.map((e) => Event(e)).toList();
  // events.forEach((e) => print(e.name));
  final events = Events();
  await events.initEvents();
  events.events.forEach((e) => print('يوم ${e.day} ${e.monthNameAr} - ${e.name}'));
}

File: events.js
const fs = require('fs');

class Event {
    constructor(json) {
        this.id = json[Event.idKey];
        this.month = json[Event.monthKey];
        this.monthNameAr = json[Event.monthNameArKey];
        this.monthNameEn = json[Event.monthNameEnKey];
        this.day = json[Event.dayKey];
        this.yearOfOccurrence = json[Event.yearOfOccurrenceKey];
        this.name = json[Event.nameKey];
        this.nameEn = json[Event.nameEnKey];
        this.description = json[Event.descriptionKey];
        this.importance = json[Event.importanceKey];
        this.typeAr = json[Event.typeArKey];
        this.typeEn = json[Event.typeEnKey];
    }

    static idKey = 'id';
    static monthKey = 'month';
    static monthNameArKey = 'month_name_ar';
    static monthNameEnKey = 'month_name_en';
    static dayKey = 'day';
    static yearOfOccurrenceKey = 'year_of_occurrence';
    static nameKey = 'name';
    static nameEnKey = 'name_en';
    static descriptionKey = 'description';
    static importanceKey = 'importance';
    static typeArKey = 'type_ar';
    static typeEnKey = 'type_en';
}

class Events {
    constructor() {
        this.events = [];
    }

    async initEvents() {
        try {
            const data = await fs.promises.readFile('events.json', 'utf8');
            const json = JSON.parse(data);
            const list = json['islamic_events'];
            this.events = list.map(e => new Event(e));
        } catch (err) {
            console.error('Error reading or parsing events.json:', err);
        }
    }
}

async function main() {
    const events = new Events();
    await events.initEvents();
    events.events.forEach(e => {
        console.log(`يوم ${e.day} ${e.monthNameAr} - ${e.name}`);
    });
}

main();
